### PR TITLE
Functionality

### PR DESCRIPTION
**Pull request by ryan-novak**
## What does this PR do?
This PR adds tests for methods on classes, as well as adds spies for DOM manipulating functions. Also adds a new class of Customer, which extends off the User class.

## Why are we doing this? Any context or related work?
To improve testing coverage for the methods on the classes, as well as being more dynamic with the addition of the Customer class.

## Where should a reviewer start?
Many many files were updated, so good luck!

### CODE REVIEW

**Comment by ryan-novak** [comment]
@vladd-png

File: src/DomUpdate.js [hunk 1]
```diff
@@ -12,61 +31,166 @@ class DomUpdate {
...
-  }
+  },
 
-  loadUser() {
+  displayDate(date) {
+    $(".date-today").text(date);
+  },
 
-  }
+  // ************************MANAGER DASHBOARD************************
```
Comment (vladd-png) [inline src/DomUpdate.js]: 💖

File: src/DomUpdate.js [hunk 2]
```diff
@@ -12,61 +31,166 @@ class DomUpdate {
...
+  //   $(".middle-section").html(`<ul class="manager-bookings-today"></ul>`);
+  //   bookingsToday.forEach(booking => {
+  //     users.find(u => )
+  //     $(".manager-bookings-today").append(`<p>${booking}</p>`)
+  //   })
+  // }
+
+    createUserSearch() {
+      $(".right-section").html(
+        `<input class="user-search-input" type="text" placeholder="Search for a user">
```
Comment (vladd-png) [inline src/DomUpdate.js]: need labels here for accessibiluity - pain

File: src/DomUpdate.js [hunk 3]
```diff
@@ -12,61 +31,166 @@ class DomUpdate {
...
+      })
+    })
+  },
+
+  createUserNewBookingButton() {
+    $(".user-new-booking-button").css("display", "block");
+  },
+
+  userNewBookingSection() {
+    this.clearSections();
```
Comment (vladd-png) [inline src/DomUpdate.js]: like this

File: src/customer.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    sortPastBookings(booking) {
+        this.pastBookings.push(booking);
+        return this.pastBookings.sort((a, b) => {
+            return b.numDate - a.numDate;
+        })
+    }
+
+    sortFutureBookings(booking) {
```
Comment (vladd-png) [inline src/customer.js]: cool idea here to sort the dates

File: src/customer.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        this.futureBookings.push(booking);
+        return this.futureBookings.sort((a, b) => {
+            return a.numDate - b.numDate;
+        })
+    }
+
+    findTotalSpent(allRooms) {
```
Comment (vladd-png) [inline src/customer.js]: this loooks great

**Comment by vladd-png** [comment]
Looks good. 👍🏼Lots of moving pieces here but you seem to have a clear direction in mind.

**Pull request merged**